kolibri/plugins/coach/assets/src/app.js

```diff
@@ -65,6 +65,12 @@ class CoachToolsModule extends KolibriApp {
         PageNames.QUIZ_SECTION_ORDER,
         PageNames.QUIZ_BOOK_MARKED_RESOURCES,
         PageNames.QUIZ_LEARNER_REPORT,
+        PageNames.LESSON_SUMMARY,
```

Comment:
We should reconsider this automatic `loading` dispatch on every redirect @marcellamaki, as this array is becoming each time more longer.

Reply:
Yes @marcellamaki, we can see how many routes are actually taking advantage of this vs how many are just skipping this. We also need to consider that in some other routes, we have this [defaultHandler](https://github.com/AlexVelezLl/kolibri/blob/ad0eeb8e4373bfd9d5f1b16260628ad07b1626dc/kolibri/plugins/coach/assets/src/routes/learnersRoutes.js#L34) that the only thing that it does is to dispatch the "notLoading" event. These pages are also basically skipping this loading state.

Reply:
yes I definitely agree with that @AlexVelezLl. We have some long standing issues with coach loading state, in terms of our overall strategy, tech debt, and also in-page loading and state management being well organized. You will see some of these in [the general maintenance milestone for 0.18](https://github.com/learningequality/kolibri/milestone/89). I think it would be productive to revive some of those conversations and make some decisions, and see how we can resolve some of these challenges in the scope of this work where possible (or create a plan for doing it in follow up). thank you for raising this!